signalfx/pyformance/registry.py

```diff
@@ -0,0 +1,194 @@
+import functools
+from .metadata import MetricMetadata
+import pyformance.registry
+import re
+import time
+from pyformance.registry import (clear, count_calls, dump_metrics,  # noqa
+                                 global_registry, meter_calls,
+                                 set_global_registry, time_calls)
+
+
+class MetricsRegistry(pyformance.registry.MetricsRegistry):
+    """An extension of the pyformance MetricsRegistry
+    which accepts and manages dimensional data to emit to SignalFx
+    """
+    def __init__(self, clock=time):
+        self.metadata = MetricMetadata()
+        super(MetricsRegistry, self).__init__(clock=clock)
+
+    def add(self, key, metric, **dims):
+        """Adds custom metric instances to the registry with dimensions
+        which are not created with their constructors default arguments
+        """
+        return super(MetricsRegistry, self).add(
+            self.metadata.register(key, **dims), metric)
+
+    def counter(self, key, **dims):
+        """adds counter with dimensions to the registry"""
+        return super(MetricsRegistry, self).counter(
+            self.metadata.register(key, **dims))
+
+    def histogram(self, key, **dims):
+        """adds histogram with dimensions to the registry"""
+        return super(MetricsRegistry, self).histogram(
+            self.metadata.register(key, **dims))
+
+    def gauge(self, key, gauge=None, default=float("nan"), **dims):
+        """adds gauge with dimensions to the registry"""
+        return super(MetricsRegistry, self).gauge(
+            self.metadata.register(key, **dims), gauge=gauge, default=default)
+
+    def meter(self, key, **dims):
+        """adds meter with dimensions to the registry"""
+        return super(MetricsRegistry, self).meter(
+            self.metadata.register(key, **dims))
+
+    def timer(self, key, **dims):
+        """adds timer with dimensions to the registry"""
+        return super(MetricsRegistry, self).timer(
+            self.metadata.register(key, **dims))
+
+    def clear(self):    # noqa flake8 complains that this is
+                        # a redefinition of the imported clear,
+                        # but obviously it isn't
+        """clears the registered metrics and metadata"""
+        self.metadata.clear()
+        super(MetricsRegistry, self).clear()
+
+
+# set global registry on import to the SignalFx MetricsRegistry
+set_global_registry(MetricsRegistry())
+
+
+class RegexRegistry(MetricsRegistry):
```

Comment:
What is this for? The documentation is copy/pasted from the other one, so I don't know how it's different

Reply:
I've updated the documentation to describe what it is used for.  Essentially it is used for pattern matching methods in REST APIs